pkg/kube/factory.go

```diff
@@ -27,19 +29,78 @@ func (f *factory) SetKubeContext(context string, config *api.Config) (*api.Confi
 	return &newConfig, nil
 }
 
+func (f *factory) SetKubeNamespace(namespace string, config *api.Config) (*api.Config, error) {
+	newConfig := *config
+	ctx := f.GetCurrentContext(config)
+	if ctx == nil {
+		return nil, errors.New("could not find Kubernetes context")
+	}
+	if ctx.Namespace == namespace {
+		return config, nil
+	}
+
+	ctx.Namespace = namespace
+	err := clientcmd.ModifyConfig(clientcmd.NewDefaultPathOptions(), newConfig, false)
+	if err != nil {
+		return nil, err
+	}
+	return &newConfig, nil
+}
+
 func NewKuber() Kuber {
 	return &factory{}
 }
 
-func (f *factory) LoadConfig() (*api.Config, error) {
+func (f *factory) LoadAPIConfig() (*api.Config, error) {
+	po := clientcmd.NewDefaultPathOptions()
+	if po == nil {
+		return nil, errors.New("unable to get kube config path options")
+	}
+	apiConfig, err := po.GetStartingConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	return apiConfig, nil
+}
+
+func (f *factory) LoadClientConfig() (*rest.Config, error) {
 	po := clientcmd.NewDefaultPathOptions()
 	if po == nil {
 		return nil, errors.New("unable to get kube config path options")
 	}
-	config, err := po.GetStartingConfig()
+	restConfig, err := clientcmd.BuildConfigFromFlags("", po.GlobalFile)
 	if err != nil {
 		return nil, err
 	}
+	// for testing purposes one can enable tracing of Kube REST API calls
+	traceKubeAPI := os.Getenv("TRACE_KUBE_API")
+	if traceKubeAPI == "1" || traceKubeAPI == "on" {
+		restConfig.WrapTransport = func(rt http.RoundTripper) http.RoundTripper {
+			return &Tracer{RoundTripper: rt}
+		}
+	}
+
+	return restConfig, nil
+}
+
+func (f *factory) GetCurrentContext(config *api.Config) *api.Context {
+	if config != nil {
+		name := config.CurrentContext
+		if name != "" && config.Contexts != nil {
+			return config.Contexts[name]
+		}
+	}
+	return nil
+}
 
-	return config, nil
+func (f *factory) GetCurrentNamespace(config *api.Config) string {
```

Comment:
Is this used?

Reply:
possibly not, should be though - think I got lost in the moment - diolch

Reply:
yeah I should be using that, I forgot about that implementation, when selecting from the list it should display the current namespace at the top - great catch bud 👍

Reply:
sorted